File: Search/LinearSearch/TypeScript/linearSearch.ts
const linearSearch = (list: number[], target: number): boolean => {
  for (let i = 0; i < list.length; i++) {
    if (list[i] === target) return true
  }

  return false
}

const newLnearSearch = (list: number[], target: number): number | Error => {
  for (let i = 0; i < list.length; i++) {
    if (list[i] === target) return i
  }

  throw 'No such value'
}
export { linearSearch, newLnearSearch }

File: Search/LinearSearch/Ruby/linear_search.rb
class NoSuchValueError < StandardError
  def initialize(msg="No such value in the array")
    super
  end
end

def liner_search(arr, target)
  i = 0
  while i < arr.length
    if arr[i] == target
      return i
    end
    i+=1
  end
  raise NoSuchValueError
end


# list = [2, 4, 6, 8, 10]
# p liner_search(list, 6) #=> 2
# p liner_search(list, 100) #=> "No such value in the array"
